functional_annotation/functional_annotation.md

```diff
@@ -0,0 +1,239 @@
+> The fees KEGG charges support curation and tool development.
+> While the KEGG database is available online, it's not fully programmitcally available unless you pay for it.
+> However, _representations_ of the data _can_ be reproduced and provided for free. 
+> kofamscan, while produced by KEGG, makes KEGG annotations available to the broader community by distributing hidden markov models of KEGG orthologs. 
+> HMMs are representations, so it's a nice workaround.
+
+
+Let's download the full assembly for SRR8859675.
+
+```
+wget -O SRR8859675_final_contigs.fasta https://osf.io/aq2nk/download
+```
+
+
+## Predicting open reading frames from bacterial and/or archaeal contigs
+
+Once we have a contig that we're interested in annotating, the first thing we need to do is identify the open reading frames.
+[ORFs](https://www.genome.gov/genetics-glossary/Open-Reading-Frame) are nucleotides that occur between start and stop codons in DNA sequences.
+ORFs often encode genes, so extracting ORFs gets us one step closer to identifying the functional content in sequences of interest.
+
+We'll use prodigal to extract open reading frames.
+As stated above, prodigal is the field-standard for ORF identification and extraction in bacterial and archaeal genomes.
+
+Run prodigal with the `-h` flag to see what options are available
+
+```
+prodigal -h
+```
+
+```
+Usage:  prodigal [-a trans_file] [-c] [-d nuc_file] [-f output_type]
+                 [-g tr_table] [-h] [-i input_file] [-m] [-n] [-o output_file]
+                 [-p mode] [-q] [-s start_file] [-t training_file] [-v]
+
+         -a:  Write protein translations to the selected file.
+         -c:  Closed ends.  Do not allow genes to run off edges.
+         -d:  Write nucleotide sequences of genes to the selected file.
+         -f:  Select output format (gbk, gff, or sco).  Default is gbk.
+         -g:  Specify a translation table to use (default 11).
+         -h:  Print help menu and exit.
+         -i:  Specify FASTA/Genbank input file (default reads from stdin).
+         -m:  Treat runs of N as masked sequence; don't build genes across them.
+         -n:  Bypass Shine-Dalgarno trainer and force a full motif scan.
+         -o:  Specify output file (default writes to stdout).
+         -p:  Select procedure (single or meta).  Default is single.
+         -q:  Run quietly (suppress normal stderr output).
+         -s:  Write all potential genes (with scores) to the selected file.
+         -t:  Write a training file (if none exists); otherwise, read and use
+              the specified training file.
+         -v:  Print version number and exit.
+```
+
+There are a few flags that are useful here.
+First, we see `-a`, which will allow us to write out amino acid sequences (translated) ORFs.
+`-d` allows us to also save the nucleotide sequences.
+Lastly, `-p` let's us tell prodigal that we're working with a metagenome where we might have fragmented contigs and partial ORFs.
```

Comment:
```suggestion
Lastly, `-p` let's us tell prodigal that we're working with a metagenome.
By default, prodigal assumes we are working with a single genome. 
In this mode, it will study the sequences we give it to try to better learn how to predict ORFs for that genome. 
By telling it we are working with a metagenome (which holds sequences from many different genomes) it will use pre-calculated training files to guide its predictions.
```